tgui/packages/tgui/interfaces/Autolathe.js

```diff
@@ -0,0 +1,272 @@
+    ''
+  );
+
+  const categories = [];
+  for (let i = 0; i < printables.length; i++) {
+    let data = printables[i];
+    if (categories.includes(data.recipe_category)) continue;
+
+    categories.push(data.recipe_category);
+  }
+
+  const [currentCategory, setCategory] = useLocalState(
+    context,
+    'current_category',
+    'All'
+  );
+
+  const filteredPrintables = printables.filter(
+    (val) =>
+      (val.recipe_category === currentCategory || currentCategory === 'All') &&
+      val.name.toLowerCase().match(currentSearch)
+  );
+
+  return (
+    <Section fill>
+      <Stack vertical fill>
+        {selectable_categories.length > 2 && ( // check that it isn't just one category and "all"
+          <Stack.Item>
+            <Tabs fluid>
+              {selectable_categories.map((val) => (
+                <Tabs.Tab
+                  selected={val === currentCategory}
+                  onClick={() => setCategory(val)}
+                  key={val}>
+                  {val}
+                </Tabs.Tab>
+              ))}
+            </Tabs>
+          </Stack.Item>
+        )}
+        <Stack.Item>
+          <Input
+            fluid
+            value={currentSearch}
+            placeholder="Search for an item"
+            onInput={(e, value) => setSearch(value.toLowerCase())}
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Flex direction="column">
+            {filteredPrintables.map((val, index) => {
+              return (
+                <Flex.Item key={index}>
+                  <Box>
+                    <Flex direction="row">
+                      <Flex.Item grow>
+                        <Button
+                          fluid
+                          content={capitalize(val.name)}
+                          disabled={!val.can_make}
+                          color={val.hidden ? 'red' : null}
+                          onClick={() =>
+                            act('make', {
+                              index: val.index,
+                              multiplier: 1,
+                            })
+                          }>
+                          {' (' + // sorry for this shitcode, also yes this will break if an autolathe uses more than 2 material types
+                            (val.materials[Object.keys(materials)[0]] &&
+                            val.materials[Object.keys(materials)[1]]
+                              ? val.materials[Object.keys(materials)[0]] +
+                              ', ' +
+                              val.materials[Object.keys(materials)[1]]
+                              : val.materials[Object.keys(materials)[0]]
+                                ? val.materials[Object.keys(materials)[0]]
+                                : val.materials[Object.keys(materials)[1]]) +
+                            ') '}
+                        </Button>
+                      </Flex.Item>
+
+                      {(val.has_multipliers && (
```

Comment:
This first bracket ( is not necessary

Reply:
eslint or prettier forces it in for some reason